### PR TITLE
2022 refresh

### PR DESCRIPTION
**Pull request by Intrepidd**
👋 

Trying to resurrect this lib

- Testing against modern ruby & activesupport versions
- Re-recorded all tests, that was a challenge as now we need to test again a real hubspot portal, before we could use the `demo` api key for POST / PUT requests but that's not allowed anymore, also some ids were hardcoded in the specs, that was fun 😓 
- Re-wrote the association API as it was using API v1 that has bugs (dissociate does not work for instance), so I had to slightly change the API

### CODE REVIEW

File: lib/hubspot/contact_list.rb [hunk 1]
```diff
@@ -92,12 +92,6 @@ def contacts(opts={})
       end
     end
 
-    # {http://developers.hubspot.com/docs/methods/lists/refresh_list}
-    def refresh
-      response = Hubspot::Connection.post_json(REFRESH_PATH, params: { list_id: @id, no_parse: true }, body: {})
```
Comment (Paul-Yves) [inline lib/hubspot/contact_list.rb:97]: `REFRESH_PATH` constant is now unused, it should be deleted

**Review by Paul-Yves — COMMENTED** [review]
Congrats for the hard work. Just a little nitpicking

File: spec/lib/hubspot/contact_list_spec.rb [hunk 2]
```diff
@@ -159,21 +194,25 @@
...
-        lists = Hubspot::ContactList.find([2,3,4])
+        lists = Hubspot::ContactList.find([list1.id,list2.id,list3.id])
         list = lists.first
         expect(list).to be_a(Hubspot::ContactList)
-        expect(list.id).to be == 2
-        expect(lists.second.id).to be == 3
-        expect(lists.last.id).to be == 4
+        expect(list.id).to be == list1.id
+        expect(lists.second.id).to be == list2.id
+        expect(lists.last.id).to be == list3.id
```
Comment (fonji) [inline spec/lib/hubspot/contact_list_spec.rb:215]: un risque qu'un jour ce ne soit pas le même ordre ?
```suggestion
        expect(lists.map(&:id)).to contain_exactly(list1.id, list2.id, list3.id)
```

**Review by fonji — COMMENTED** [review]
Bon je pinaille un peu, mais en soit, y a que la constante à nettoyer qui me freine à envoyer ça.
C'est grave si je commente les PRs en français ?

File: lib/hubspot/contact_list.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (fonji) [reply lib/hubspot/contact_list.rb:97]: (tiens ça fait répétition avec le commentaire de Paul-Yves :sweat_smile: )

**Review by fonji — APPROVED** [review]

**Review by Paul-Yves — APPROVED** [review]

File: lib/hubspot/contact_list.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Intrepidd) [reply lib/hubspot/contact_list.rb:97]: Oui l'endpoint n'existe plus à priori ...

**Pull request merged**